api/src/test/java/org/apache/iceberg/AssertHelpers.java

```diff
@@ -23,7 +23,22 @@
 import org.apache.avro.generic.GenericRecord;
 import org.assertj.core.api.AbstractThrowableAssert;
 import org.assertj.core.api.Assertions;
+import org.assertj.core.api.ThrowableAssert;
 
+/**
+ * This class is deprecated. Please use {@link
+ * Assertions#assertThatThrownBy(ThrowableAssert.ThrowingCallable)} directly as shown below:
+ *
+ * <pre>
+ * Assertions.assertThatThrownBy(() -> throwingCallable)
+ *    .isInstanceOf(ExpectedException.class)
+ *    .hasMessage(expectedErrorMsg)
+ * </pre>
+ *
+ * @deprecated Use {@link Assertions#assertThatThrownBy(ThrowableAssert.ThrowingCallable)} directly
+ *     as it provides a more fluent way of asserting on exceptions.
```

Comment:
My only issue with this is making sure that `hasMessage` is called. How do we enforce that?

Reply:
that's a valid point. I was thinking maybe we could introduce a checkstyle rule (a multiline regex) that would make sure checks against the error message are done. I can open a separate PR for that.

Reply:
I opened https://github.com/apache/iceberg/issues/7040 to address that

Reply:
all calling sites that use any method from `AssertHelpers` will see that the class itself is deprecated, so I don't think we need to deprecate each method individually

Reply:
I did some small tests, there is a small difference in IDE: if only the static method is imported the user won't see deprecation warning in code unless he/she looks at the import block, so it might still be beneficial to mark methods as deprecated as well.

Reply:
good point @jackye1995. I've added deprecations to all methods inside that class